File: backend/Microservices/AccountsMicroservice/Accounts.DataAccess/Settings/PaginationSettings.cs
namespace Accounts.DataAccess.Settings
{
    public class PaginationSettings
    {
        private const int maxPageSize = 50;
        private int _pageSize = 5;

        public int PageNumber { get; set; } = 1;

        public int PageSize
        {
            get
            {
                return _pageSize;
            }
            set
            {
                _pageSize = (value > maxPageSize) ? maxPageSize : value;
            }
        }
    }
}

File: frontend/src/app/settings/pagination-settings.ts
export class PaginationSettings {
    private readonly maxPageSize = 50;
    private _pageSize = 5;
  
    pageNumber = 1;
  
    get pageSize(): number {
      return this._pageSize;
    }
  
    set pageSize(value: number) {
      this._pageSize = value > this.maxPageSize ? this.maxPageSize : value;
    }
}